php/class-fieldmanager-media.php

```diff
@@ -154,22 +154,69 @@ public function presave( $value, $current_value = array() ) {
 	public function form_element( $value = array() ) {
 		if ( is_numeric( $value ) && $value > 0 ) {
 			$attachment = get_post( $value );
-			if ( strpos( $attachment->post_mime_type, 'image/' ) === 0 ) {
-				$preview  = esc_html( $this->selected_image_label ) . '<br />';
-				$preview .= '<a href="#">' . wp_get_attachment_image(
-					$value,
-					$this->preview_size,
-					false,
-					array(
-						'class' => $this->thumbnail_class,
-					)
-				) . '</a>';
-			} else {
-				$preview  = esc_html( $this->selected_file_label ) . '&nbsp;';
-				$preview .= wp_get_attachment_link( $value, $this->preview_size, true, true, $attachment->post_title );
+			// Open the preview wrapper.
+			$preview = '<div class="media-file-preview">';
+			$file_label = ''; // The uploaded file label - image or file.
+			$dashicon_class = '';
+
+			if ( $attachment->post_mime_type ) {
+				$preview .= '<a href="#">';
+				$image_type = strpos( $attachment->post_mime_type, 'image/' );
+
+				if ( strpos( $attachment->post_mime_type, 'audio/' ) === 0 ) {
+					$dashicon_class = 'dashicons-media-audio';
+				} elseif ( strpos( $attachment->post_mime_type, 'video/' ) === 0 ) {
+					$dashicon_class = 'dashicons-media-video';
+				} else {
+					$dashicon_class = 'dashicons-media-document';
+				}
```

Comment:
`wp_attachment_is()` might be preferable here compared to checking the `post_mime_type` directly, as it will allow filters such as `wp_audio_extensions` to be used for sites allowing additional MIME types.